rfcs/2021-01-26-crypto.md

```diff
@@ -18,17 +18,48 @@ One of the consideration is to use [noise protocol framework](https://noiseproto
 
 During TCP session both client and server should use symmetric AES 256 bit encryption using the session key that will be established during the handshake.
 
-To establish the session key, the server should have an asymmetric key pair generated during server deployment and unknown to the clients. The users should know the key fingerprint (hash of the public key) in advance.
+To establish the session key, the server should have an asymmetric key pair generated during server deployment and unknown to the clients. The users should know the key hash (256 bits) and additional server ID (256 bits) in advance in order to be able to establish connection.
 
 The handshake sequence could be the following:
 
 1. Once the connection is established, the server sends its public key to the client
 2. The client compares the hash of the received key with the hash it already has (e.g. received as part of connection invitation or server in NEW command). If the hash does not match, the client must terminate the connection.
-3. If the hash is the same, the client should generate a random symmetric AES key that will be used as a session key both by the client and the server.
-4. The client then should encrypt this symmetric key with the public key that the server sent and send it back to the server
-5. The server should decrypt the received symmetric key with its private key and in case of success confirm to the client that it is now ready to receive the commands.
-6. In case of decryption failure, the server should notify the client and terminate the connection.
-7. All the subsequent data both from the client and from the server should be sent encrypted.
+3. If the hash is the same, the client should generate a random symmetric AES key and IV that will be used as a session key both by the client and the server.
+4. The client then should encrypt this symmetric key with the public key that the server sent and send back to the server the result and the server ID also shared with the client in advance: `rsa-encrypt(aes-key, iv, server-id)`.
+5. The server should decrypt the received key, IV and server id with its private key.
+6. The server should compare the `server-id` sent by the client and if it does not match its ID terminate the connection.
+7. In case of successful decryption and matching key, the server should send encrypted welcome header.
+
+```abnf
+aes_welcome_header = aes_header_auth_tag aes_encrypted_header
```

Comment:
aes_encrypted_header -> welcome_header?

Reply:
or there's a transition missing

Reply:
got it, I guess. You're skipping aes encryption is only implied by names, correct? at least name this aes_welcome_header?

Reply:
I mean I understand what you're doing here.. I just think it would be more easy to follow if skipped transitions were given at least as comments